pyomo/contrib/cp/tests/test_logical_to_disjunctive.py

```diff
@@ -283,6 +307,138 @@ def test_at_least(self):
+            len(
+                list(
+                    m.disjuncts[0].component_data_objects(Disjunct, descend_into=False)
+                )
+            ),
+            0,
+        )
+
+        # nested
+        self.assertEqual(
+            len(
+                list(
+                    m.disjuncts[1].component_data_objects(
+                        Constraint, descend_into=False
+                    )
+                )
+            ),
+            0,
+        )
+        self.assertEqual(
+            len(
+                list(
+                    m.disjuncts[1].component_data_objects(Disjunct, descend_into=False)
+                )
+            ),
+            2,
+        )
+        self.assertEqual(len(m.disjuncts[1].disjunction.disjuncts), 2)
+        assertExpressionsEqual(
+            self, m.disjuncts[1].disjunction.disjuncts[0].constraint[1].expr, c + d <= 0
+        )
+        assertExpressionsEqual(
+            self, m.disjuncts[1].disjunction.disjuncts[1].constraint[1].expr, c + d >= 2
+        )
+
+        self.assertEqual(len(m.disjunctions), 1)
+        self.assertIs(m.disjunctions[0].disjuncts[0], m.disjuncts[0])
+        self.assertIs(m.disjunctions[0].disjuncts[1], m.disjuncts[1])
+
+        self.assertEqual(len(m.cons), 17)
+        assertExpressionsEqual(self, m.cons[1].expr, (1 - z3) + a + b >= 1)
+        assertExpressionsEqual(self, m.cons[2].expr, z3 + (1 - a) >= 1)
+        assertExpressionsEqual(self, m.cons[3].expr, z3 + (1 - b) >= 1)
+        assertExpressionsEqual(self, m.cons[4].expr, z4 == 1 - z3)
+        assertExpressionsEqual(self, m.cons[5].expr, z5 <= z4)
+        assertExpressionsEqual(
+            self, m.cons[6].expr, z5 <= m.disjuncts[0].binary_indicator_var
+        )
+        assertExpressionsEqual(
+            self,
+            m.cons[7].expr,
+            1 - z5 <= 2 - (z4 + m.disjuncts[0].binary_indicator_var),
+        )
+        assertExpressionsEqual(self, m.cons[8].expr, (1 - z6) + (1 - t) + z5 >= 1)
+        assertExpressionsEqual(self, m.cons[9].expr, 1 - (1 - t) + z6 >= 1)
+        assertExpressionsEqual(self, m.cons[10].expr, z6 + (1 - z5) >= 1)
+        assertExpressionsEqual(self, m.cons[11].expr, (1 - z7) + (1 - z5) + t >= 1)
+        assertExpressionsEqual(self, m.cons[12].expr, z7 + (1 - t) >= 1)
+        assertExpressionsEqual(self, m.cons[13].expr, 1 - (1 - z5) + z7 >= 1)
+        assertExpressionsEqual(self, m.cons[14].expr, z8 <= z6)
+        assertExpressionsEqual(self, m.cons[15].expr, z8 <= z7)
+        assertExpressionsEqual(self, m.cons[16].expr, 1 - z8 <= 2 - (z6 + z7))
+        assertExpressionsEqual(self, m.cons[17].expr, z8 >= 1)
+
+        TransformationFactory('gdp.bigm').apply_to(m)
+        m.obj = Objective(expr=m.t.get_associated_binary())
+        SolverFactory('gurobi').solve(m, tee=True)
+        update_boolean_vars_from_binary(m)
+
+        self.assertTrue(value(e))
+        self.assertEqual(value(m.obj), 1)
+        self.assertTrue(value(m.t))
+
```

Comment:
Pretty ugly test, indeed. Does it need Gurobi to be solved?!

Reply:
Probably there's a smarter way... But the real check is that the model can't cheat and set `m.t` to `False`. So easiest is just to minimize the associated binary and make sure it's forced True.